source/particles/particle_handler.cc

```diff
@@ -898,7 +898,7 @@ namespace Particles
         {
           const int ierr = MPI_Irecv(&(n_recv_data[i]),
                                      1,
-                                     MPI_INT,
+                                     MPI_UNSIGNED,
```

Comment:
What about `Utilities::::MPI::internal::mpi_type_id(neighbors.data())`?

Reply:
not exactly pleasant to read in my opinion but it has the advantage that it will stay correct...

Reply:
Part of #7303 was a wrapper around `MPI_Irecv` and `MPI_Isend`. Maybe, it is worth to turn that into a convenience wrapper that uses the correct data type automatically.